extensions/amp-story-interactive/1.0/utils.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Action} from '../../amp-story/1.0/amp-story-store-service';
+import {map} from '../../../src/utils/object';
+import {user} from '../../../src/log';
+
+export const updateInteractiveStoreState = (storeService, interactive) => {
+  const update = {
+    'option': interactive.getOptionSelected(),
+    'interactiveId': interactive.getInteractiveId(),
+  };
+  storeService.dispatch(Action.ADD_INTERACTIVE_REACT, update);
+};
+
+/** @private Whether ids are deduplicated or not */
+let deduplicatedIds = false;
+
+/**
+ * Deduplicates the interactive Ids, only called once
+ * @param {!Document} doc
+ */
+export const deduplicateInteractiveIds = (doc) => {
```

Comment:
You need to call this before registering each component in the store?

Reply:
I call it the first time that I call getInteractiveId so the Ids are guaranteed to be unique. Ready for review